### PR TITLE
TASK: Remove legacy `NodeAddress` and `NodeAddressFactory`

### PR DESCRIPTION
**Pull request by mhsdesign**
Neos Ui adjustments https://github.com/neos/neos-ui/pull/3858
Fixes final parts of https://github.com/neos/neos-development-collection/issues/4564

As bitterly debated in https://github.com/neos/neos-development-collection/issues/5072 we will change the serialisation format of the CR Node in URLS and in the Neos UI.

Currently there is a big legacy corruption layer around the WorkspaceModule, NeosUi and the Nodes controller using the ancient Legacy NodeAddress and NodeAddressFactory to work with the legacy format. This is part of the migration strategy and will be cleaned up in this change.

For reference the Neos Ui's content node serialisation:

```
- Neos 8.3
onedimension.localhost:8081/neos/content?node=/sites/neos-test-site@user-admin;language=en_US

- Neos 9.0-dev (currently)
onedimension.localhost:8081/neos/content?node=user-admin__eyJsYW5ndWFnZSI6ImVuX1VTIn0=__f676459d-ca77-44bc-aeea-44114814c279

- Neos 9.0 next beta, this pr
onedimension.localhost:8081/neos/content?node={"contentRepositoryId":"onedimension","workspaceName":"user-admin","dimensionSpacePoint":{"language":"en_US"},"aggregateId":"000b42ff-c9d3-4b5e-b6de-56d35832dc0e"}
```

**Upgrade instructions**

**Review instructions**

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/Classes/Controller/Service/NodesController.php [hunk 1]
```diff
@@ -238,7 +236,7 @@ public function showAction(string $identifier, string $workspaceName = 'live', a
             }
         });
 
-        $nodeAddress = NodeAddressFactory::create($contentRepository)->createFromNode($node)->serializeForUri();
+        $nodeAddress = NodeAddress::FromNode($node)->toJson();
```
Comment (mhsdesign) [inline Neos.Neos/Classes/Controller/Service/NodesController.php]:
```suggestion
        $nodeAddress = NodeAddress::fromNode($node)->toJson();
```

File: Neos.Workspace.Ui/Classes/Controller/WorkspaceController.php [hunk 2]
```diff
@@ -836,16 +814,16 @@ protected function computeSiteChanges(Workspace $selectedWorkspace, ContentRepos
                     // As for changes of type `delete` we are using nodes from the live content stream
                     // we can't create `serializedNodeAddress` from the node.
```
Comment (mhsdesign) [inline Neos.Workspace.Ui/Classes/Controller/WorkspaceController.php]: outdated.

File: Neos.Neos/Classes/Controller/Backend/ContentController.php [hunk 3]
```diff
@@ -147,12 +146,9 @@ public function initializeUploadAssetAction(): void
     public function uploadAssetAction(Asset $asset, string $metadata, string $node, string $propertyName)
     {
         $nodeAddressString = $node;
```
Comment (pKallert) [inline Neos.Neos/Classes/Controller/Backend/ContentController.php]: $nodeAddressString is only assigned to pass it on the $nodeAddress, perhaps we can just do 
$nodeAddress = NodeAddress::fromJsonString($node);

**Review by pKallert — APPROVED** [review]
Tested it locally and works fine as far as I can see 👍

**Review by bwaidelich — APPROVED** [review]
lgtm, thanks!

**Review by kitsunet — APPROVED** [review]

**Comment by mhsdesign** [comment]
Thanks for the reviews <3 
I also merged the Neos Ui adjustments: https://github.com/neos/neos-ui/pull/3858

**Pull request merged**